soc/nordic/nrf54h/soc.c

```diff
@@ -29,6 +31,12 @@ LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
 #define HSFLL_NODE DT_NODELABEL(cpurad_hsfll)
 #endif
 
+#if !(defined(CONFIG_PM) || defined(CONFIG_POWEROFF))
+#if !defined(CONFIG_SOC_NRF54H20_CPURAD)
+static sys_snode_t soc_node;
+#endif
+#endif
+
```

Comment:
Instead of these `#if defined`s, how about a function like `const * sys_snode_t soc_act_pd_sys_snode_get(void)` and define this sys_snode_t always? This would also invert the dependencies: from pm to soc - like it should be.